packages/legend-query-builder/src/components/result/tds/QueryBuilderTDSResultShared.tsx

```diff
@@ -394,28 +639,87 @@ const filterByOrOutValue = (
       tdsState,
     );
   } else {
-    generateNewPostFilterConditionNodeData(
+    try {
+      await generateNewPostFilterConditionNodeData(
+        applicationStore,
+        operator,
+        _cellData,
+        tdsState,
+        tdsColState,
+      );
+    } catch (error) {
+      assertErrorThrown(error);
+      applicationStore.alertUnhandledError(error);
+    }
+  }
+};
+
+const filterByOrOutValue = async (
+  applicationStore: ApplicationStore<
+    LegendApplicationConfig,
+    LegendApplicationPluginManager<LegendApplicationPlugin>
+  >,
+  isFilterBy: boolean,
+  _cellData: QueryBuilderTDSResultCellData,
+  data: QueryBuilderTDSResultCellData | null,
+  tdsState: QueryBuilderTDSState,
+): Promise<void> => {
+  const tdsColState = data?.columnName
+    ? getTDSColumnState(tdsState, _cellData.columnName)
+    : _cellData?.columnName
+      ? getTDSColumnState(tdsState, _cellData.columnName)
+      : undefined;
```

Comment:
This is the fix for the bug that was occurring previously. When the enterprise grid is used, the value passed in for `data` is `null` on the first click, which was causing us to not get the `tdsColState` correctly. To fix this, we can use the value passed in for `_cellData` as a fallback, which does exist on the first grid click.